db/migrate/20230615002110_add_marketplace_order_notification_methods.rb

```diff
@@ -0,0 +1,10 @@
+class AddMarketplaceOrderNotificationMethods < ActiveRecord::Migration[7.0]
+  def change
+    create_table :marketplace_order_notification_methods, id: :uuid do |t|
+      t.references :marketplace, type: :uuid, foreign_key: {to_table: :furnitures}
+      t.string :contact_method, default: :email, null: false
+      t.string :contact_location_ciphertext, null: false
```

Comment:
why `contact_location` and not `contact_email`? Does the app expect to have different contact types?

Reply:
I'm guessing Zee is thinking maybe one day this could be e.g. a phone number.

Reply:
At some point, I would love for us to have a `ContactMethod` within the higher-level Convene Operating System to support this kind of thing; so that each `Gizmo` or other use-case doesn't have to re-invent it every time.

However, I haven't taken the time to actually think more about it or implement it... But basically I imagine there being several fields:

- `#via` (Is this XMPP? Email? SMS? Signal? Voice Call? Physical Mail? WhatsApp? Twitter? ActivityPub?)
- `#location` (the actual address/phone number/etc.)
- `#validated_by` (which `Person` validated the `ContactMethod`)
- `#validated_on` (when the `Person` validated the `ContactMethod`)

Theoretically, this would get squeezed out of the `AuthenticationMethod` class; so that a `Person` can have many `ContactMethod`s with *some* of those `ContactMethod`s being appropriate for authenticating that the person is who they say they are.